src/teuthology_api/services/kill.py

```diff
@@ -4,60 +4,72 @@
 
 from fastapi import HTTPException, Request
 
-from teuthology_api.services.helpers import get_username, get_run_details
+from teuthology_api.services.helpers import get_username, get_run_details, isAdmin
 
 
 TEUTHOLOGY_PATH = os.getenv("TEUTHOLOGY_PATH")
+ADMIN_TEAM = os.getenv("ADMIN_TEAM")
+
 log = logging.getLogger(__name__)
 
 
-def run(args, send_logs: bool, access_token: str, request: Request):
+async def run(args, send_logs: bool, token: dict, request: Request):
     """
     Kill running teuthology jobs.
     """
-    if not access_token:
+    if not token:
         log.error("access_token empty, user probably is not logged in.")
         raise HTTPException(
             status_code=401,
             detail="You need to be logged in",
             headers={"WWW-Authenticate": "Bearer"},
         )
     username = get_username(request)
-    run_name = args.get("--run")
+    run_name = args.get("--run", "")
     if run_name:
         run_details = get_run_details(run_name)
-        run_owner = run_details.get("user")
+        jobs_details = run_details.get("jobs", [])
+        if jobs_details:
+            run_owner = jobs_details[0].get("owner", "")
     else:
         log.error("teuthology-kill is missing --run")
         raise HTTPException(status_code=400, detail="--run is a required argument")
-    # TODO if user has admin priviledge, then they can kill any run/job.
-    if run_owner.lower() != username.lower():
-        log.error(
-            "%s doesn't have permission to kill a job scheduled by: %s",
-            username,
-            run_owner,
-        )
-        raise HTTPException(
-            status_code=401, detail="You don't have permission to kill this run/job"
-        )
+
+    if (run_owner.lower() != username.lower()) and (
+        run_owner.lower() != f"scheduled_{username.lower()}@teuthology"
+    ):
+        isUserAdmin = await isAdmin(username, token["access_token"])
```

Comment:
just a little nit, if token dic exists but the "access_token" key doesn't exist, are we handling this properly?